### DISCUSSION TITLE
breaking grammar change on main branch between 7mo ago (mar 9th 2024) and now

### DISCUSSION THREAD

**Discussion by BenGardiner** [strictdoc-project/strictdoc#1960] [Q&A]
Hi strictdoc! Apologies but the automated builds at https://github.com/nmfta-repo/vcr-experiment/ got disable due to inactivity. So I don't have a very tight timeframe on this breakage. 

last working build https://github.com/nmfta-repo/vcr-experiment/actions/runs/8218066306

I looked over the commits from Mar 10th to today and I couldn't clearly identify what broke it. I _can_ setup a git bisect of it and will do if  required (i.e. if no one here can help me guess how to fix my sdoc files).

Maybe it has something to do with included docs or perhaps https://github.com/strictdoc-project/strictdoc/commit/2a30c994a86283666eaed0fe10b26170a94f695f  ?

broken build: https://github.com/nmfta-repo/vcr-experiment/actions/runs/11521687710/job/32075900513#step:7:154

**Comment by BenGardiner** [comment]
sorry I meant to merge @stanislaw 's prev fix first. the breakage is here https://github.com/nmfta-repo/vcr-experiment/actions/runs/11522087794/job/32077178654#step:7:95

**Reply by stanislaw** [reply]
@BenGardiner the REFS are RELATIONS now, see https://strictdoc.readthedocs.io/en/latest/strictdoc_01_user_guide.html#relations-previously-refs. This should fix your SDoc files.

**Reply by BenGardiner** [reply]
You're a 🏆 champion thanks @stanislaw !

**Reply by BenGardiner** [reply]
sadly it wasn't as simple as `git ls-files "*.sdoc" | xargs sed -i 's/\bREFS\b/RELATIONS/g'` I'm still debugging

**Comment by BenGardiner** [comment]
fixing the build breakage required also:
1. putting all relations as the last element in each requirement AND
2. repeating the `[GRAMMAR]` block in all of the included fragments.

**Reply by BenGardiner** [reply]
for reference https://github.com/nmfta-repo/vcr-experiment/pull/59

**Comment by stanislaw** [comment]
This should be solved by now. Closing the discussion.
